pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -185,6 +199,20 @@ func (o createAuthInfoOptions) validate() error {
 	if len(methods) > 1 {
 		return fmt.Errorf("You cannot specify more than one authentication method at the same time: %v", strings.Join(methods, ", "))
 	}
+	if o.embedCertData.Value() {
+		if o.clientCertificate.Value() == "" {
+			return errors.New("You must specify a client certificate to embed")
+		}
+		if o.clientKey.Value() == "" {
```

Comment:
Requiring that a user provide both client cert and client key when using --embed-certs, but still allowing specifying only one or the other when specifying them as files seems a little odd.  Perhaps we could always require them to be set as a pair in all cases (you may specify neither or both, but not just one)?